packages/manager/apps/web/client/app/hosting/cdn/shared/hosting-cdn-shared-settings.controller.js

```diff
@@ -0,0 +1,348 @@
+      tasks.push(
+        this.HostingCdnSharedService.updateCDNDomainOption(
+          serviceName,
+          domainName,
+          HOSTING_CDN_SHARED_SETTINGS_RULE_TYPE_DEVMODE,
+          {
+            type: HOSTING_CDN_SHARED_SETTINGS_RULE_TYPE_DEVMODE,
+            enabled: this.model.devmode.enabled,
+          },
+        ),
+      );
+
+    if (this.model.brotli.changed)
+      tasks.push(
+        this.HostingCdnSharedService.updateCDNDomainOption(
+          serviceName,
+          domainName,
+          HOSTING_CDN_SHARED_SETTINGS_RULE_TYPE_BROTLI,
+          {
+            type: HOSTING_CDN_SHARED_SETTINGS_RULE_TYPE_BROTLI,
+            enabled: this.model.brotli.enabled,
+          },
+        ),
+      );
+
+    return tasks;
+  }
+
+  getSentRulesToAdd() {
+    if (this.tasks.toAdd.length) {
+      const { serviceName, domainName } = this.params;
+
+      return map(this.tasks.toAdd, (rule) => {
+        return this.HostingCdnSharedService.addNewOptionToDomain(
+          serviceName,
+          domainName,
+          rule,
+        );
+      });
+    }
+
+    return [];
+  }
+
+  getSentRulesToUpdate() {
+    if (this.tasks.toUpdate.length) {
+      const { serviceName, domainName } = this.params;
+
+      return map(this.tasks.toUpdate, (rule) => {
+        const cRule = clone(rule, true);
+        delete cRule.name;
+        return this.HostingCdnSharedService.updateCDNDomainOption(
+          serviceName,
+          domainName,
+          rule.name,
+          cRule,
+        );
+      });
+    }
+
+    return [];
+  }
+
+  getSentRulesToRemove() {
+    if (this.tasks.toRemove.length) {
+      const { serviceName, domainName } = this.params;
+
+      return map(this.tasks.toRemove, (rule) => {
+        return this.HostingCdnSharedService.deleteCDNDomainOption(
+          serviceName,
+          domainName,
+          rule.name,
+        );
+      });
+    }
+
+    return [];
+  }
+
+  static getChangedSwitches(model) {
+    const { alwaysOnline, http, devmode, brotli } = model;
+    const switchesTasks = [alwaysOnline, http, devmode, brotli];
+    return filter(switchesTasks, (s) => s && s.changed);
+  }
+
+  static getChangedRules(tasks) {
+    const { toAdd, toUpdate, toRemove } = tasks;
+    return concat(toAdd, toRemove, toUpdate);
+  }
+
+  openConfirmModal() {
+    const switches = this.constructor.getChangedSwitches(this.model);
+    const rules = this.constructor.getChangedRules(this.tasks);
+    if (switches.length || rules.length) {
+      this.$state.go('app.hosting.cdn.shared.confirmSettings', {
+        rules,
+        domain: this.params.domain,
+        model: this.model,
+        success: () => this.saveSettings(),
+        cancel: () => {},
+      });
+    }
+  }
+
+  saveSettings() {
+    const tasks = concat(
+      this.getSentSwitchesOption(),
+      this.getSentRulesToRemove(),
+      this.getSentRulesToUpdate(),
+      this.getSentRulesToAdd(),
+    );
+    this.$q
+      .all(tasks)
```

Comment:
As we could have 2 HTTP calls for switches options (`devmod`/`brotli`), X for `RulesToRemove`, Y for `RulesToUpdate` and Z for `RulesToAdd`(2 + X + Y + Z).
Is there some limitation (max rules, etc) ? it could mean a lot of HTTP calls...
What's happen if one call fails ? Partial configuration applied and that's all ?

Reply:
yes i had same reflexion, but according the spec it will happen like so.
yes changes will be saved individually and if error occurred it will be displayed.